### PR TITLE
st: vl6180x: Driver for vl6180x

### PR DESCRIPTION
**Pull request by benys**
The vl6180x is time of flight sensor. Sensor publish proximity and distance channel.

Signed-off-by: Kamil Serwus <kserwus@gmail.com>

### CODE REVIEW

File: samples/sensor/vl6180x/README.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
+.. _vl6180x:
+
+VL6180X: Time Of Flight sensor
```
Comment (MaureenHelm) [inline samples/sensor/vl6180x/README.rst:3]: Please make this sample generic to any TOF sensor and add it in a separate commit

File: samples/sensor/vl6180x/prj.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_I2C=y
+CONFIG_GPIO=y
+CONFIG_SENSOR=y
+CONFIG_VL6180X_PROXIMITY_THRESHOLD=100
+
+#debug
+#CONFIG_DEBUG=y
+#CONFIG_LOG=y
+#CONFIG_SENSOR_LOG_LEVEL_DBG=y
```
Comment (MaureenHelm) [inline samples/sensor/vl6180x/prj.conf:10]: Remove

File: samples/sensor/vl6180x/prj.conf [hunk 3]
```diff
@@ -0,0 +1,11 @@
+CONFIG_STDOUT_CONSOLE=y
+CONFIG_I2C=y
```
Comment (MaureenHelm) [inline samples/sensor/vl6180x/prj.conf:2]: This is redundant and can be removed

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/vl6180x/vl6180x.c [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+		ret = VL6180_SetI2CAddress(&drv_data->vl6180x, 2 * config->i2c.addr);
+		if (ret != 0) {
+			LOG_ERR("[%s] Unable to reconfigure I2C address",
+				dev->name);
+			return -EIO;
+		}
+
+		drv_data->vl6180x.I2cDevAddr = config->i2c.addr;
+		LOG_DBG("[%s] I2C address reconfigured", dev->name);
+		k_sleep(K_MSEC(2));
```
[the remark this answers is not in the document]
Comment (benys) [reply drivers/sensor/vl6180x/vl6180x.c:80]: Datasheet (https://www.st.com/resource/en/datasheet/vl6180x.pdf) said that device is ready after 1ms 400us.
So I assumes that after set new address we need to wait. 
But I see at documentation that is any timeing information after this operation. I remove this and I test it.

File: samples/sensor/vl6180x/README.rst (hunk 1 above)
Comment (benys) [reply samples/sensor/vl6180x/README.rst:3]: @MaureenHelm 
Can you give me a little help - can you point me another generic sample?

For example:
So I has to change from vl6180x to tof? And define some other overloads?

I don't udnerstand :-(

Comment (MaureenHelm) [reply samples/sensor/vl6180x/README.rst:3]: See `samples/sensor/accel_polling` and `samples/sensor/magn_polling`

**Comment by seyednasermoravej** [comment]
Hi,
Why this changes aren't on the main Zephyr repository?
How can I use this driver?
Thanks in advance

**Pull request closed without merging**